cookbooks/bach_krb5/libraries/utils.rb

```diff
@@ -0,0 +1,18 @@
+def principal_exists?(pname) 
+  require 'rkerberos'
+  kadm5_connected = false
+  princ_found = false
+
+  begin
+    kadm5 = Kerberos::Kadm5.new(:principal => "#{node[:krb5][:admin_principal]}", :password => "#{get_config("krb5-admin-password")}")
```

Comment:
Same point about `get_config!` instead of `get_config` to raise if the password is not found.